src/server/pfs/server/driver.go

```diff
@@ -1997,22 +1975,45 @@ func (d *driver) subscribeCommit(pachClient *client.APIClient, repo *pfs.Repo, b
 		case watch.EventError:
 			return event.Err
 		case watch.EventPut:
-			if err := event.Unmarshal(&branchName, branchInfo); err != nil {
+			if err := event.Unmarshal(&commitID, commitInfo); err != nil {
 				return fmt.Errorf("Unmarshal: %v", err)
 			}
-			if branchInfo.Head == nil {
-				continue // put event == new branch was created. No commits yet though
+			if commitInfo == nil {
+				return fmt.Errorf("Commit info is empty for id: %v", commitID)
+			}
+
+			// if provenance is provided, ensure that the returned commits have the commit in their provenance
+			if prov != nil {
+				valid := false
+				for _, cProv := range commitInfo.Provenance {
+					if cProv.Commit.Repo.Name == prov.Commit.Repo.Name &&
+						cProv.Commit.ID == prov.Commit.ID &&
+						cProv.Branch.Name == prov.Branch.Name {
```

Comment:
Assuming you're just comparing all the fields of these protos to each other you can just do it with `proto.Equal` to make it a bit shorter. And it'll mean that new fields automatically get included in the comparison, which seems unlikely to matter here, but you never know.

Reply:
oh nice, i didn't know about that!